ivy/functional/frontends/torch/linalg.py

```diff
@@ -94,3 +94,33 @@ def svd(A, /, *, full_matrices=True, driver=None, out=None):
 def svdvals(A, *, driver=None, out=None):
     # TODO: add handling for driver
     return ivy.svdvals(A, out=out)
+
+
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"1.11.0 and below": ("float16", "bfloat16")}, "torch")
+def tensorinv(input, ind=2):
+    input_shape = ivy.shape(input)
+    if ind > 0:
+        shape_ind_end = input_shape[:ind]
+        shape_ind_start = input_shape[ind:]
+        prod_ind_end = 1
+        prod_ind_start = 1
+        for i in shape_ind_start:
+            prod_ind_start *= i
+        for j in shape_ind_end:
+            prod_ind_end *= j
+        if prod_ind_end == prod_ind_start:
+            inverse_shape = shape_ind_start + shape_ind_end
+            input = ivy.reshape(input, shape=(prod_ind_end, -1))
+            inverse_shape_tuple = tuple([*inverse_shape])
+            if len(ivy.shape(input)) > 1:
+                inverse_tensor = ivy.inv(input)
+            else:
+                return ivy.reshape(input, shape=inverse_shape_tuple)
+        else:
+            raise ValueError(
+                            "Tensor shape must satisfy prod"
+                            "(A.shape[:ind]) == prod(A.shape[ind:])")
+    else:
+        raise ValueError("Expected a strictly positive integer for 'ind'")
+    return ivy.reshape(inverse_tensor, shape=inverse_shape_tuple)
```

Comment:
There is a special case that I think needs handling. When `A` is a 2-dimensional tensor and `ind=1`, this function should return `torch.linalg.inv(A)`. This is a note in the [documentation](https://pytorch.org/docs/stable/generated/torch.linalg.tensorinv.html).

Reply:
I think 2-dim tensor is basically a matrix and to satisfy product of shape dimensions equals till given axis(in our case it's 1, so the shape must be square) and product of rest, it should keep the shape same. Because `inverse_shape` is the same of `input_shape` in this case.
(e.g.: `shape = (n, n) and ind = 1`) 
In case it's not square shaped and `ind=1`, will throw an error. Imho, that's why the original c++ implementation has no handling code block for this case. 
If you still think that should be added, I can do that. However, I feel like it cause code redundancy.